### PR TITLE
Save file id, file name, and project code in usage log

### PR DESCRIPTION
**Pull request by adecler**
### NOTE: Depends on 
https://github.com/BHoM/BHoM_Engine/pull/2312

   
### Issues addressed by this PR
Provide changes required for fileName and project code to be saved by analytics. Two things were required here:
 - A new component that can be used to associate a script with a specific project
 - new fields in the usage log entries to store file id, file name and project code

A few things to note:
- I use a new type of event that allows to easily have the project code available when processing the log entries. It also serves as a nice way to let the user know that the component did what it was supposed to do.
- I saved file id because the script doesn't have a file name until it is saved. And, at that point, there is no guarantee that the existing components will be ran again. In case there are multiple files opened in a session, it is critical to be able to safely identify the script owning each component.
- I used `projectCode` instead of `projectNumber` (and therefore a string instead of a number) because there is no guarantee that other companies will also use a number.

### Test files
See [BHoMAnalytics PR](https://github.com/BHoM/BHoMAnalytics_Toolkit/pull/16)

### CODE REVIEW

**Comment by adecler** [comment]
@BHoMBot check copyright

**Comment by adecler** [comment]
@BHoMBot check installer

**Review by al-fisher — CHANGES_REQUESTED** [review]
System working well.
Suggested changes below.

Tested in GH and Excel with other comments given https://github.com/BHoM/Excel_Toolkit/pull/280#pullrequestreview-592432620 and https://github.com/BHoM/Grasshopper_Toolkit/pull/604#pullrequestreview-592600402
Thanks @adecler

File: UI_Engine/Compute/SetProjectCode.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public static partial class Compute
+    {
+        /*************************************/
+        /**** Public Methods              ****/
+        /*************************************/
+
+        [Description(@"Set the code of the project this file is related to.")]
+        [Input("projectCode", "Project code/number this file is associated with.")]
+        [Output("success", "Returns true if the project code was correctly registered.")]
+        public static bool SetProjectCode(string projectCode)
```
[the remark this answers is not in the document]
Comment (al-fisher) [reply UI_Engine/Compute/SetProjectCode.cs:45]: And then actually playing with the tooltips on the canvas - might be worth keeping the verbose _"this script is associated with."_ on both descriptions of method name and input like you had 

```suggestion
        [Description(@"Set the project number or unique project ID this script is associated with.")]
        [Input("projectID", "Project number/unique project ID this script is associated with.")]
        [Output("success", "Returns true if the project ID was correctly registered.")]
        public static bool SetProjectID(string projectID)
```

**Comment by adecler** [comment]
@BHoMBot check copyright

**Comment by adecler** [comment]
@BHoMBot check installer

**Review by al-fisher — APPROVED** [review]
Changes look good. Thanks @adecler

**Pull request merged**